app/Http/Controllers/api/StudentCollaborationController.php

```diff
@@ -3,20 +3,31 @@
 namespace App\Http\Controllers\api;
 
 use App\Http\Controllers\Controller;
+use App\Service\CollaborationService;
+use Exception;
+use Illuminate\Database\Eloquent\ModelNotFoundException;
 
 //use Illuminate\Http\Request;
 
 class StudentCollaborationController extends Controller
 {
-    public function __invoke()
-    {
-
-        $data = file_get_contents(base_path('database/data/studentCollaborations.json'));
-
-        $decodedData = json_decode($data, true);
-
-        return response()->json($decodedData);
-
+    protected $collaborationService;
 
+    public function __construct(CollaborationService $collaborationService)
+    {
+        $this->collaborationService = $collaborationService;
+    }
+    public function __invoke($uuid)
+    {
+        try {
+            $collaborationDetail = [
+                'collaborations' => $this->collaborationService->getCollaborationDetails($uuid),
+            ];
+            return response()->json($collaborationDetail);
+        } catch (ModelNotFoundException $e) {
+            return response()->json(['error' => 'Student not found.'], 404);
+        } catch (Exception $e) {
+            return response()->json(['error' => 'Internal server error.'], 500);
+        }
     }
```

Comment:
I agree with Rosa's comment.

You could take advantage of the exceptions already created, ``StudentNotFoundException`` and ``ResumeNotFoundException``, in your ``__invoke()`` function.

And quoting a comment from Jordi on another PR: _"Controllers are the place where we do check that the things in our domain are working fine and where we do catch the exceptions to return controlled responses."_

As a suggestion of that, something like this:

~~~
public function __invoke($uuid)
    {
        try {
            $collaborationDetail = [
                'collaborations' => $this->collaborationService->getCollaborationDetails($uuid),
            ];
            return response()->json($collaborationDetail);
        } catch (StudentNotFoundException $e) {
            return response()->json(['message' => $e->getMessage()], $e->getCode());
        } catch (ResumeNotFoundException $e) {
            return response()->json(['message' => $e->getMessage()], $e->getCode());
        } catch (\Exception $e) {
            return response()->json(['message' => $e->getMessage()], $e->getCode() ?: 500);
        }
    }
~~~